3Dmol/glmodel.js

```diff
@@ -2455,25 +2455,31 @@ $3Dmol.GLModel = (function() {
     * @function $3Dmol.GLModel#setCoordinatesFromURL
     * @param {string} url - contains the url where mdsrv has been hosted
     * @param {string} path - contains the path of the file (<root>/filename)
+    * @param {function} callback - function called after request
     */
 
-        this.setCoordinatesFromURL = function* (url, path) {
+        this.setCoordinatesFromURL = function (url, path, callback) {
             var atomCount = atoms.length;
             frames = [];
-
-            var numFrames = yield function(){return new Promise(function(resolve,reject){
-                $.get("http://"+url+"/traj/numframes/"+path,resolve);
-            })}();
-
-            if (!isNaN(parseInt(numFrames))) {
-                frames.push(atoms);
-                frames.numFrames = numFrames;
-                frames.url = url;
-                frames.path = path; 
-            }
-
-            yield this.setFrame(0);
-
+            var self = this;
```

Comment:
If we're using promises, then the function should return a promise.  It should not take a callback argument.  The user does something like:
```
setCoordinatesFromURL(url).then(callback);
```